packages/server/src/node/launch/websocket-server-launcher.ts

```diff
@@ -0,0 +1,157 @@
+    @inject(Logger)
+    protected override logger: Logger;
+
+    protected server: Server;
+    protected startupCompleteMessage = START_UP_COMPLETE_MSG;
+    protected currentConnections: jsonrpc.MessageConnection[] = [];
+
+    protected async run(options: WebSocketServerOptions): Promise<void> {
+        const resolvedOptions = await this.resolveOptions(options);
+        this.server = new Server({ server: resolvedOptions.server, path: resolvedOptions.path });
+        const endpoint = `ws://${resolvedOptions.host}:${resolvedOptions.port}:${resolvedOptions.path}`;
+        this.logger.info(`The GLSP Websocket launcher is ready to accept new client requests on endpoint '${endpoint}'`);
+        console.log(this.startupCompleteMessage.concat(resolvedOptions.port.toString()));
+
+        this.server.on('connection', (ws, req) => {
+            this.createClientConnection(ws);
+        });
+
+        return new Promise((resolve, reject) => {
+            this.server.on('close', () => resolve(undefined));
+            this.server.on('error', error => reject(error));
+        });
+    }
+
+    protected async createClientConnection(socket: WebSocket): Promise<void> {
+        const container = this.createContainer();
+        const connection = this.createConnection(socket);
+        this.currentConnections.push(connection);
+        const glspServer = container.get<JsonRpcGLSPServer>(JsonRpcGLSPServer);
+        glspServer.connect(connection);
+        this.logger.info('Starting GLSP server connection');
+        connection.listen();
+        connection.onDispose(() => this.disposeClientConnection(container, glspServer));
+        socket.on('close', () => this.disposeClientConnection(container, glspServer));
+        connection.onClose(() => console.log('GOD dam'));
+        return new Promise((resolve, rejects) => {
+            connection.onClose(() => resolve(undefined));
+            connection.onError(error => rejects(error));
+        });
+    }
+
+    protected disposeClientConnection(container: Container, glspServer: GLSPServer): void {
+        glspServer.shutdown();
+        container.unbindAll();
+    }
+
+    protected createConnection(socket: WebSocket): jsonrpc.MessageConnection {
+        return createWebSocketConnection(wrapWebSocket(socket));
+    }
+
+    protected async resolveOptions(options: WebSocketServerOptions): Promise<ResolvedWebSocketServerOptions> {
+        const port = !options.port || options.port === 0 ? await getFreePort() : options.port;
+        const path = options.path.startsWith('/') ? options.path : `/${options.path}`;
+        const server = this.createHttpServer(port, options.host);
+
+        return { ...options, port, path, server };
+    }
+
+    protected createHttpServer(port: number, host?: string): http.Server {
+        const server = http.createServer((req, res) => {
+            const body = http.STATUS_CODES[426];
```

Comment:
I had to look up that status code so I think it would be great to either import some package that already provides them or at least create a constant for the one we use `const STATUS_UPGRADE_REQUIRED = 426` or something similar.